src/main/java/com/minecrafttas/tasmodog/container/InputContainer.java

```diff
@@ -0,0 +1,158 @@
+package com.minecrafttas.tasmodog.container;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.lwjgl.input.Keyboard;
+
+import com.minecrafttas.tasmodog.TASmod;
+import com.minecrafttas.tasmodog.tools.KeyboardHelper;
+
+/**
+ * Input container containing inputs and other tasmod data for all ticks
+ * @author Pancake
+ */
+public class InputContainer {
+
+	private TASmod tasmod;
+	private List<Tick> ticks;
+	private Tick currentTick;
+	private int nextTick;
+	private boolean isActive, isRecording;
```

Comment:
One mistake I made when doing TASmod was having a lot of booleans dictate the states...
This comes down to preference of course but I suggest making an enum with 3 states.  

```java
public enum States{
NONE,PLAYBACK,RECORD;
}
```
Implementing state changes is so much easier that way and you can add new states later if they are ever required. For me this was the `Paused` state that was active during loading screens

Reply:
Good point. I have not thought about that.